include/textslider.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the QPxTool project.
+ * Copyright (C) 2009 Gennady "ShultZ" Kozlov <qpxtool@mail.ru>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file "COPYING" for the exact licensing terms.
+ *
+ */
+
+#ifndef _TEXT_SLIDER_H
+#define _TEXT_SLIDER_H
+
+#include <QAbstractSlider>
+#include <QStyleOptionSlider>
+
+class SliderItem {
+public:
+  SliderItem(QString s, bool en = true) {
+    text = s;
+    enabled = en;
+  }
+
+  inline bool operator==(SliderItem o) { return text == o.text; };
+  inline bool operator==(SliderItem &o) { return text == o.text; };
+  inline bool operator==(QString o) { return text == o; };
+  inline bool operator==(QString &o) { return text == o; };
+
+  QString text;
+  bool enabled;
+};
+
+class TextSlider : public QAbstractSlider {
+  Q_OBJECT
+public:
+  TextSlider(QWidget *p = NULL);
+  TextSlider(Qt::Orientation, QWidget *p = NULL);
```

Comment:
Codacy found an issue: [Class 'TextSlider' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)